### PR TITLE
feat(kumactl): delete ebpf resources from all the nodes 

### PR DESCRIPTION
**Pull request by lukidzi**
###
Fixes: https://github.com/kumahq/kuma-net/issues/78

Missing things:
- removing from the VM
- add kuma-net code that gonna remove files and unpin programs, and probably tc when on VM
- tests

### Checklist prior to review

- [ ] Link to docs PR or issue --
- [ ] Link to UI issue or PR --
- [ ] Is the [issue worked on linked][1]? --
- [ ] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [ ] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Unit Tests --
- [ ] E2E Tests --
- [ ] Manual Universal Tests --
- [ ] Manual Kubernetes Tests --
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [ ] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch)?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: app/kumactl/cmd/install/context/install_control_plane_context.go [hunk 1]
```diff
@@ -123,6 +124,7 @@ func DefaultInstallCpContext() InstallCpContext {
 			Egress_enabled:                          false,
 			Egress_drainTime:                        "30s",
 			Egress_service_type:                     "ClusterIP",
+			Ebpf_bpffspath:                          "/run/kuma/bpf",
```
Comment (lahabana) [inline app/kumactl/cmd/install/context/install_control_plane_context.go:127]: We agreed on not adding more params here and simply rely on regular values from helm instead.

File: app/kumactl/cmd/uninstall/uninstall_ebpf.go [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+}
+
+func newUninstallEbpf(root *kumactl_cmd.RootContext) *cobra.Command {
+	args := ebpfArgs{
+		// default value that we inject in pod injector
+		BPFFsPath:           root.InstallCpContext.Args.Ebpf_bpffspath,
+		Timeout:             time.Duration(120 * time.Second),
+		CleanupImageVersion: kuma_version.Build.Version,
+		RemoveOnly:          false,
+		Namespace:           root.InstallCpContext.Args.Namespace,
```
Comment (lahabana) [inline app/kumactl/cmd/uninstall/uninstall_ebpf.go:57]: I wouldn't bother with reusing the root context here.

File: app/kumactl/cmd/uninstall/uninstall_ebpf.go [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	typedbatchv1 "k8s.io/client-go/kubernetes/typed/batch/v1"
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+const (
+	AppLabel          = "kuma.io/bpf-cleanup"
+	BpfCleanupJobName = "kuma-bpf-cleanup"
+	BpfCleanupImage   = "kumahq/kuma-init"
```
Comment (lahabana) [inline app/kumactl/cmd/uninstall/uninstall_ebpf.go:28]: You need to make this configurable as often people will rename image or move them in different registries

File: deployments/charts/kuma/templates/post-delete-cleanup-ebpf-job.yaml [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+      {{ toYaml . | nindent 8 }}
+      {{- end }}
+      restartPolicy: OnFailure
+      {{- if .Values.hooks.podSecurityContext }}
+      securityContext:
+      {{ toYaml .Values.hooks.podSecurityContext | trim | nindent 8 }}
+      {{- end }}
+      containers:
+        - name: post-delete-job
+          image: "kumahq/kuma-init:0.0.0-preview.4583fbf95"
```
Comment (lahabana) [inline deployments/charts/kuma/templates/post-delete-cleanup-ebpf-job.yaml:102]: Need to make both image and version configurable.

**Comment by lahabana** [comment]
@lukidzi Is #5095 replacing this?

**Comment by lukidzi** [comment]
> @lukidzi Is #5095 replacing this?

Yes, I am going to close this

**Pull request closed without merging**